packages/test/test-end-to-end-tests/src/test/handleValidation.spec.ts

```diff
@@ -0,0 +1,349 @@
+	{
+		type: registerId,
+		storeHandle: async (defaultDataStore, handle) => {
+			const registerRoot = await defaultDataStore.getSharedObject(registerId);
+			registerRoot.write("B", handle);
+		},
+		readHandle: async (defaultDataStore) => {
+			const registerRoot = await defaultDataStore.getSharedObject(registerId);
+			return registerRoot.read("B") as IFluidHandle<ITestFluidObject>;
+		},
+	},
+	{
+		type: queueId,
+		storeHandle: async (defaultDataStore, handle) => {
+			const queueRoot = await defaultDataStore.getSharedObject(queueId);
+			queueRoot.add(handle);
+		},
+		readHandle: async (defaultDataStore) => {
+			let handle2: unknown;
+			const queueRoot = await defaultDataStore.getSharedObject(queueId);
+			const callback: ConsensusCallback<IFluidHandle> = async (value) => {
+				handle2 = value;
+				return ConsensusResult.Complete;
+			};
+			await queueRoot.acquire(callback);
+			return handle2;
+		},
+	},
+	// {
+	// 	type: migrationShimId,
+	// 	storeHandle: async (defaultDataStore, handle) => {
+	// 		const migrationShimRoot = await defaultDataStore.getSharedObject(migrationShimId);
+	// 		const tree = migrationShimRoot.currentTree as LegacySharedTree;
+	// 		const legacyNodeId: TraitLabel = "inventory" as TraitLabel;
+
+	// 		const handleNode: BuildNode = {
+	// 			definition: legacyNodeId,
+	// 			traits: {
+	// 				handle,
+	// 			},
+	// 		};
+	// 		tree.applyEdit(
+	// 			Change.insertTree(
+	// 				handleNode,
+	// 				StablePlace.atStartOf({
+	// 					parent: tree.currentView.root,
+	// 					label: legacyNodeId,
+	// 				}),
+	// 			),
+	// 		);
+
+	// 		const rootNode = tree.currentView.getViewNode(tree.currentView.root);
+	// 		// eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+	// 		const nodeId = rootNode.traits.get(legacyNodeId)![0];
+	// 		const change = Change.setPayload(nodeId, handle);
+	// 		tree.applyEdit(change);
+	// 	},
+	// },
+];
+
+describeCompat("handle validation", "NoCompat", (getTestObjectProvider, apis) => {
+	for (const handle of handleFns) {
+		it(`store handle in dds: ${handle.type}`, async () => {
+			const { container1, provider, testContainerConfig } = await setup(
+				getTestObjectProvider,
+				apis,
+			);
+			const defaultDataStore = (await container1.getEntryPoint()) as ITestFluidObject;
+			const runtime = defaultDataStore.context.containerRuntime;
+
+			const dataStoreB = await runtime.createDataStore(["default"]);
+			const dataObjectB = (await dataStoreB.entryPoint.get()) as ITestFluidObject;
+			const idB = dataObjectB.context.id;
+
+			await handle.storeHandle(defaultDataStore, dataObjectB.handle);
+
+			await provider.ensureSynchronized();
+			const container2: IContainerExperimental =
+				await provider.loadTestContainer(testContainerConfig);
+			await waitForContainerConnection(container2);
+			const default2 = (await container2.getEntryPoint()) as ITestFluidObject;
+
+			const handleB = (await handle.readHandle(default2)) as IFluidHandle<ITestFluidObject>;
```

Comment:
we should avoid casting. in the stress test we use isFluidHandle, we should use that here like `assert(isFuildHandle(handle2))` rather than casting